app/models/user.rb

```diff
@@ -19,7 +24,13 @@ def social_worker?
     role == User::SOCIAL_WORKER
   end
 
-  validates :role, inclusion: { in: ROLES }
+  def department_manager?
+    role == User::DEPARTMENT_MANAGER
+  end
+
+  def superadmin_or_department_manager?
```

Comment:
Doing this check in every policy method over and over again just has the potential typo risk raised. Mixing up `||` or anything anywhere is maybe unlikely, but not impossible.

The naming may not be totally pleasant. I chose it like that for having the method completely self explanatory.